packages/trader/src/Modules/Trading/Components/Form/ContractType/contract-type-item.tsx

```diff
@@ -2,32 +2,39 @@ import classNames from 'classnames';
 import React from 'react';
 import { Text } from '@deriv/components';
 import IconTradeCategory from 'Assets/Trading/Categories/icon-trade-categories';
-import { TContractType } from './types';
+import { TContractType, TFilteredContractType } from './types';
 
 type TItem = {
-    contract_types: TContractType[];
+    contract_types: TFilteredContractType[];
     handleSelect?: (clicked_item: TContractType, e: React.MouseEvent<HTMLDivElement>) => void;
     value?: string;
 };
 
 const Item = ({ contract_types, handleSelect, value }: TItem) => (
     <React.Fragment>
-        {contract_types.map(type => (
-            <div
-                id={`dt_contract_${type.value}_item`}
-                data-testid='dt_contract_item'
-                key={type.value}
-                className={classNames('contract-type-item', {
-                    'contract-type-item--selected': value === type.value || value?.includes(type.value),
-                })}
-                onClick={e => handleSelect?.(type, e)}
-            >
-                <IconTradeCategory category={type.value} className='contract-type-item__icon-wrapper' />
-                <Text size='xs' className='contract-type-item__title'>
-                    {type.text}
-                </Text>
-            </div>
-        ))}
+        {contract_types.map(type => {
+            const tradeType = Array.isArray(type.value)
```

Comment:
instead of doing all this cant we just create a regex to see if contract_type string has turbo or vanilla?

Reply:
You mean something like:
```
const value_substitute = value;
if value === turbosshort, then value_substitute = turboslong;
if value === vanillaput, then value_substitute = vanillashort;
```
`/then, instead of all this Array.isArray checks, we just substitute checks with value_substitute/`

Seems shorter as for me

Reply:
@henry-deriv @kate-deriv I turned `type.value` into an array for `vanillas`, `turbos` and `rise_fall/rise_fall_equal`.  And this list can be extended later in case more new trade types will be added. I'm doing all these `if` checks for specific trade types inside `reduce` in the parent (`contract-type-tist.tsx`) already.
I honestly believe that there is no point in checking for specific trade types here for the second time because having these checks in one place (in the parent) is enough. This child component doesn't really need to know anything about specific trade types.

>use a cached value of a selected_contract_type
`value` is the selected trade type.

What we do in this child is: if `type.value` is array, we check whether or not the array includes the value, otherwise we simply keep the old check whether or not the `value` is equal to `type.value`. It's the simplest way to handle this.

Reply:
yeah i mean the whole point is to check if the contract_type has vanilla, turbo, or risefall using regex. Then if theres a match then dont change selected_contract_type